### ISSUE TITLE
Extend `GetCallMetadata` with indices

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#13511]
`GetCallMetadata` can currently be used to get the call- and module-name of a `RuntimeCall`.  
In cases where we want to select specific calls/modules, it is better to use their index than strings.  

`GetCallMetadata` could therefore be extended to also return these indices. Looking internally, we use a `GetCallName` trait which is re-used in the `GetCallMetadata` trait. So probably analogous to that add a `GetCallIndex` and then re-use that as well.

**Comment by kziemianek** [comment]
Hi, I would love to do it :)

**Comment by ggwpez** [comment]
Okay great! yes please go ahead

**Comment by kziemianek** [comment]
@ggwpez, I think I have most of stuff done, but I'm struggling to figure out how to get function index from `decl_module!` macro (`frame/support/src/dispatch.rs`). I need this to fill `GetCallIndex` trait implementation. Could you guide me a little bit ?

**Comment by ggwpez** [comment]
Yes best to open a Draft MR @kziemianek, then it is easier to collaborate and comment.

**Comment by kziemianek** [comment]
@ggwpez I've opened, please take a look.

**Comment by liamaharon** [comment]
Closed in https://github.com/paritytech/substrate/pull/13558

**Comment by ggwpez** [comment]
> Closed in #13558

That was just the first part. The second part is blocked on removing the old `decl_*` pallet macros.
